gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -924,6 +931,146 @@ public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOExce
     }
   }
 
+  /** See {@link GoogleCloudStorage#deleteFolders(List)} for details about expected behavior. */
+  @Override
+  public void deleteFolders(List<FolderInfo> folders) throws IOException {
+    logger.atFiner().log("deleteFolders(%s)", folders);
+
+    if (folders.isEmpty()) {
+      return;
+    }
+
+    // Gather exceptions to wrap in a composite exception at the end.
+    final KeySetView<IOException, Boolean> innerExceptions = ConcurrentHashMap.newKeySet();
+    String traceContext = String.format("batchFolderDelete(size=%s)", folders.size());
+    try (ITraceOperation to = TraceOperation.addToExistingTrace(traceContext)) {
+
+      // Map to store number of occurrence of each parent Folder name
```

Comment:
Consider rewording this as number of children for a given folder?

Reply:
Updated